### PR TITLE
#232 added column cards count to pie chart

### LINKED ISSUE #232: Upgrade the pie chart shown in src/components/regions/UserDashboard/UserDetailedStats.js so that it includes a count of cards in each column.
Blocked by #231 . Currently, not all the information is available. Please don't start this card until 231 is merged to the correct place

Please update the story and fixtures to demonstrate that this works

Please use feat/user-cart-stats-pie-chart as a base branch

### PR DESCRIPTION
**Pull request by babalwa-mbolekwa**
Related issues: [#232 ]

## Description:

What are you up to? Fill us in :)

## Screenshots/videos
piechart_cardscount

## I solemnly swear that:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by FaithMo — APPROVED** [review]
Looks good

File: frontend/src/components/regions/UserDashboard/UserDetailedStats.js [hunk 1]
```diff
@@ -46,24 +46,30 @@ const renderCustomizedLabel = ({ cardStatusPieData }) => {
 
 export default ({ detailedStats }) => {
   if (!detailedStats) return <React.Fragment />;
+  
+  const cardsAssignedWithStatusComplete = detailedStats.cardsAssignedWithStatusComplete;
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDashboard/UserDetailedStats.js]: Please make use of destructuring.

File: frontend/src/components/regions/UserDashboard/UserDetailedStats.js [hunk 2]
```diff
@@ -46,24 +46,30 @@ const renderCustomizedLabel = ({ cardStatusPieData }) => {
...
+  const cardsAssignedWithStatusInReview = detailedStats.cardsAssignedWithStatusInReview;
+  const cardsAssignedWithStatusReviewFeedback = detailedStats.cardsAssignedWithStatusReviewFeedback;
+  const cardsAssignedWithStatusInProgress = detailedStats.cardsAssignedWithStatusInProgress;
+
   const cardStatusPieData = [
-    { name: "Complete", value: detailedStats.cardsInCompletedColumnAsAssignee },
-    { name: "Review", value: detailedStats.cardsInReviewColumnAsAssignee },
+    { name: `Complete (${cardsAssignedWithStatusComplete})`, value: cardsAssignedWithStatusComplete },
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDashboard/UserDetailedStats.js]: Please make sure all the different card statuses are covered. Include Blocked.

**Review by sheenarbw — CHANGES_REQUESTED** [review]

**Comment by babalwa-mbolekwa** [comment]
piechart

**Review by FaithMo — APPROVED** [review]

**Review by Vuyisanani — CHANGES_REQUESTED** [review]
Please resolve your merge conflicts first

**Review by Sbonelo01 — APPROVED** [review]

**Review by Vuyisanani — APPROVED** [review]

File: frontend/src/components/regions/UserDashboard/utils.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * fillColor function returns a colour string value based on the type of column:
+ * Grey for Blocked column,
+ * Blue for Ready column,
+ * Green for In Progress column,
+ * Red for Review Feedback column,
+ * Orange for Review column,
+ * Yellow for Complete column
+ */
+
+export const fillColor = (columnName) => {
```
Comment (sheenarbw) [inline frontend/src/components/regions/UserDashboard/utils.js]: This seems a bit overkill. There is a much simpler way. If we just treat data as data then you dont even need a function to work this stuff out.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
It looks like it works, but the code can be much simpler.  This problem is not a very complicated one, it doesn't need this much spagetti. 

If you have a piece of code that needs some knowledge then try to find the most straightforward way to make that knowledge available.  For me to see how the colours are derived I shouldn't need to go on a treasure hunt.

**Review by Vuyisanani — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**